File: python/function/is_even.py
"""
Write a short Python function, is_even(k), that takes an integer value and
returns True if k is even, and False otherwise. However, your function cannot use
the multiplication, modulo, or division operators.
"""

def is_even(k: int) -> bool:
    if k & 1 == 0:
        return True
    else:
        return False

File: typescript/function/is_even.ts
/*
* 2. Write a short Python function, `is_even(k)`, that takes an integer value and
* return True if k is even, and False otherwise. However, your function cannot use
* the multiplication, module, or division operators.
*
*/

export function is_even(k: number): boolean {
    if ((k & 1) === 0) {
        return true;
    }
    else {
        return false;
    }
}